test/types.test.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Attributes,
+  AttributeValue,
+  JSONMLNode,
+} from '..';
+
+import * as jsonml from '..';
+
+// Utils
+const isFragYes: boolean = jsonml.isFragment(['', 'hi']);
+const isFragNo: boolean = jsonml.isFragment(['div', 'hi']);
+
+const paraTag: string = jsonml.getTagName(['p', 'some text']);
+const fragTag: string = jsonml.getTagName(['', 'some fragment text']);
+
+const foo = jsonml.getTagName([
+  '', // fragment
+  { foo: 'bar' }, // attributes allowed here
+  ['p', { foo: 'bar' }, 'some fragment text'],
+  ['p', { foo: 'bar' }, 'some fragment text'],
+  ['div', 'foo'],
+  'baz'
+]);
```

Comment:
Tested and can verify that I get issues if the attributes are moved to the wrong position or if an `undefined` or otherwise invalid element is added somewhere.